src/window.c

```diff
@@ -3472,13 +3873,29 @@ __editor_maybe_static int file_manager(struct world *mzx_world,
 int choose_file_ch(struct world *mzx_world, const char *const *wildcards,
  char *ret, const char *title, int dirs_okay)
 {
+  int slot_result = SLOTSEL_FILE_MANAGER_RESULT;
+
+  if(get_config()->save_slots && strcmp(*wildcards, ".SAV") == 0)
+    slot_result = slot_manager(mzx_world, ret, title, false);
+
+  if(slot_result != SLOTSEL_FILE_MANAGER_RESULT)
+    return slot_result;
+
   return file_manager(mzx_world, wildcards, NULL, ret, title, dirs_okay,
    0, NULL, 0, 0);
 }
 
 int new_file(struct world *mzx_world, const char *const *wildcards,
  const char *default_ext, char *ret, const char *title, int dirs_okay)
 {
+  int slot_result = SLOTSEL_FILE_MANAGER_RESULT;
+
+  if(get_config()->save_slots && strcmp(*wildcards, ".SAV") == 0)
+    slot_result = slot_manager(mzx_world, ret, title, true);
+
+  if(slot_result != SLOTSEL_FILE_MANAGER_RESULT)
+    return slot_result;
+
```

Comment:
The hooks for these definitely *do not* belong here. `load_savegame_selection` in game.c and the `IKEY_s` case in `game_key` are better candidates for this. I think `slot_manager`/etc. is okay being in window.c though.

Reply:
Yeah, this really shouldn't have stuck around this long. I intended to work my way back to all of the functions that handle game saving/loading and by the time I finished the stuff in window.c I forgot why I left my original TODO in the first place. :P Shouldn't be hard to fix this.

Reply:
Corrected in 1c21352.